x/qgb/orchestrator/evm_client.go

```diff
@@ -86,7 +89,22 @@ func (ec *evmClient) UpdateValidatorSet(
 	if err != nil {
 		return err
 	}
-	ec.logger.Info("ValSetUpdate", tx.Hash().String())
+
+	// TODO put this in a separate function and listen for new EVM blocks instead of just sleeping
+	for i := 0; i < 60; i++ {
+		ec.logger.Debug(fmt.Sprintf("waiting for valset %d to be confirmed: %s", nonce, tx.Hash().String()))
+		lastNonce, err := ec.StateLastValsetNonce(&bind.CallOpts{Context: ctx})
+		if err != nil {
+			return err
+		}
+		if lastNonce == nonce {
+			ec.logger.Info(fmt.Sprintf("relayed valset %d: %s", nonce, tx.Hash().String()))
+			return nil
+		}
+		time.Sleep(10 * time.Second)
+	}
```

Comment:
as stated in the todo, I like the idea of a separate evm client method that waits for a specific update or something similar

Reply:
let's write that down in an issue if we haven't already tho

Reply:
We can create a generic method that subscribes to new blocks and checks if a transaction hash is included or not.
Could be something like:
```golang
func (ec *EvmClient) waitForConfirmations(txHash string, numberOfConfirmations int) error
```
I wrote some snippets, but it would take much time debugging and making it work. Especially that subscribing to events requires using a web socket connection and that needs to be added to the config which itself needs to be updated to reflect the different commands.
Thus, I decided to leave it for later and do this instead.

Reply:
issue: https://github.com/celestiaorg/celestia-app/issues/356

Reply:
:shipit: